src/app/course/[courseId]/board/page.tsx

```diff
@@ -0,0 +1,93 @@
+"use client";
+
+import Board from "@components/board";
+import { useOfficeHour } from "@hooks/oh/useOfficeHour";
+import { IdentifiableQuestions } from "@interfaces/type";
+import { Box, Button } from "@mui/material";
+import CheckIcon from "@mui/icons-material/Check";
+import React from "react";
+
+interface PageProps {
+  params: {
+    courseId: string;
+  };
+}
+
+const SELECT_ALL = "All";
+
+const Page = (props: PageProps) => {
+  const { course, questions } = useOfficeHour();
+  const [selectedTopics, setSelectedTopics] = React.useState([SELECT_ALL]);
+
+  const topics = [SELECT_ALL, ...Object.keys(course.tags).sort()];
+
+  const getQuestionsByTopic = (topics: string[]) => {
+    if (topics.length === 0) {
+      return [];
+    } else if (topics.includes(SELECT_ALL)) {
+      return questions;
+    } else {
+      return questions.filter((question) =>
+        question.tags.find((tag) => topics.includes(tag))
+      );
+    }
+  };
+
+  const addOrRemoveTopic = (topic: string) => {
+    setSelectedTopics((topics) => {
+      if (topics.includes(topic)) {
+        return topics.filter((other) => other !== topic);
+      } else {
+        return [...topics, topic];
+      }
+    });
+  };
+
+  return (
+    <Box
+      paddingX="16px"
+      display="flex"
+      flexDirection="column"
+      rowGap="16px"
+      height="100vh"
+    >
+      <Box display="flex" justifyContent="space-between">
+        <h1 style={{ fontSize: "32px", color: "#191C20" }}>Board</h1>
+        <Button sx={{ color: "#38608F" }}>View History</Button>
+      </Box>
+      <Box display="flex" columnGap="16px" overflow="auto" paddingBottom="16px">
+        {topics.map((topic) => (
+          <Button
+            key={topic}
+            sx={{
+              flexShrink: 0,
+              padding: "8px 16px",
+              color: "#1D192B",
+              whiteSpace: "nowrap",
+              minWidth: "content",
+              display: "flex",
+              alignItems: "center",
+              justifyContent: "center",
+              backgroundColor: selectedTopics.includes(topic) ? "#D7E3F8" : "",
```

Comment:
Weird bug with the topic background color selection. The topic states should be fine since the checkmark is being rendered conditionally, possibly an under the hood bug with the backgroundColor prop?

https://github.com/user-attachments/assets/233d9630-fdf2-4a5d-83c0-23df2f4a2935

Reply:
It's lagging behind because there's a hover effect on MUI button. If you're on a non-mobile screen, it looks fine. I wonder if it works as expected on actual phone

Reply:
Yea, seems like this a :hover bug with mui button with some browsers. (omitted the mui github issue link so our work doesn't auto-link to their issue). There doesn't seem to be a fix with responsive view on chrome, but this should work on the actual phone!